tests/acceptance/features/bootstrap/UserLdapContext.php

```diff
@@ -0,0 +1,287 @@
+	 * We need to make sure that the setup routines are called in the correct order
+	 * So this is the main function for setUp
+	 * 
+	 * @param BeforeScenarioScope $scope
+	 * @return void
+	 */
+	public function setUpBeforeScenario(BeforeScenarioScope $scope) {
+		$suiteParameters = SetupHelper::getSuiteParameters($scope);
+		$this->connectToLdap($suiteParameters);
+		$this->importLdifFile(
+			__DIR__ . (string)$suiteParameters['ldapInitialUserFilePath']
+		);
+		$this->theLdapUsersHaveBeenResyncted();
+	}
+
+	/**
+	 * @param array $suiteParameters
+	 * @return void
+	 */
+	public function connectToLdap($suiteParameters) {
+		$this->ldapAdminUser = (string)$suiteParameters['ldapAdminUser'];
+		$this->ldapAdminPassword = (string)$suiteParameters['ldapAdminPassword'];
+		$this->ldapBaseDN = (string)$suiteParameters['ldapBaseDN'];
+		$this->ldapUsersOU = (string)$suiteParameters['ldapUsersOU'];
+		$this->ldapGroupsOU = (string)$suiteParameters['ldapGroupsOU'];
+		$ci = getenv("CI");
+		if ($ci === "drone") {
+			$this->ldapHost = (string)$suiteParameters['ldapHostDrone'];
+		} else {
+			$this->ldapHost = (string)$suiteParameters['ldapHost'];
+		}
+		
+		$options = array(
+			'host' => $this->ldapHost,
+			'password' => $this->ldapAdminPassword,
+			'bindRequiresDn' => true,
+			'baseDn' => $this->ldapBaseDN,
+			'username' => $this->ldapAdminUser
+		);
+		$this->ldap = new Zend\Ldap\Ldap($options);
+		$this->ldap->bind();
+		SetupHelper::init(
+			"admin", (string)$suiteParameters['adminPassword'],
+			$this->getMinkParameter('base_url'), $suiteParameters['ocPath']
+		);
+	}
+
+	/**
+	 * imports an ldif string
+	 * 
+	 * @param string $ldifData
+	 * 
+	 * @return void
+	 */
+	public function importLdifData($ldifData) {
+		$items = Zend\Ldap\Ldif\Encoder::decode($ldifData);
+		
+		if (isset($items['dn'])) {
+			//only one item in the ldif data
+			$this->ldap->add($items['dn'], $items);
+		} else {
+			foreach ($items as $item) {
+				$this->ldap->add($item['dn'], $item);
+			}
+		}
+	}
+
+	/**
+	 * 
+	 * @param string $path
+	 * @return void
+	 */
+	public function importLdifFile($path) {
+		$ldifData = file_get_contents($path);
+		$this->importLdifData($ldifData);
+	}
+
+	/**
+	 * @AfterScenario
+	 * @return void
+	 */
+	public function deleteUserAndGroups() {
+		$this->ldap->delete(
+			"ou=" . $this->ldapUsersOU . "," . $this->ldapBaseDN, true
+		);
+		$this->ldap->delete(
+			"ou=" . $this->ldapGroupsOU . "," . $this->ldapBaseDN, true
+		);
+		$this->theLdapUsersHaveBeenResyncted();
+	}
+
+	/**
+	 * After Scenario. Sets back old settings
+	 *
+	 * @return void
+	 * @AfterScenario
+	 */
+	public function resetOldConfig() {
+		foreach ($this->oldConfig as $configId => $settings ) {
+			foreach ($settings as $configKey => $configValue) {
+				$this->setLdapSetting($configId, $configKey, $configValue);
+			}
+		}
+	}
+
+	/**
+	 * 
+	 * @param string $configId
+	 * @param string $configKey
+	 * @param string $configValue
+	 * @throws Exception
```

Comment:
While we are getting started, might as well put a blank line between the ``param`` tags and the other ``return`` and ``throws`` tags. (done last week in core)